scipio/src/executor.rs

```diff
@@ -302,7 +346,7 @@ impl ExecutorQueues {
 /// ```
 /// use scipio::LocalExecutor;
 ///
-/// let local_ex = LocalExecutor::new(None).expect("failed to create local executor");
+/// let local_ex = LocalExecutor::spawn_default();
```

Comment:
I do not like the name `spawn` for the version that does not create a new thread.

In all of rust when you `spawn` something you are given a `JoinHandle` to wait on. But not here. 
Let's go with `build`, `create` or anything else that does not mention spawn

Reply:
Agreed, new naming scheme:
* `make` variants returns the `LocalExecutor`
* `spawn` variants spawn threads and return `JoinHandle`s